sympy/integrals/prde.py

```diff
@@ -918,14 +918,25 @@ def parametric_log_deriv_heu(fa, fd, wa, wd, DE, c1=None):
 
 
 def parametric_log_deriv(fa, fd, wa, wd, DE):
-    # TODO: Write the full algorithm using the structure theorems.
-#    try:
-    A = parametric_log_deriv_heu(fa, fd, wa, wd, DE)
-#    except NotImplementedError:
-        # Heuristic failed, we have to use the full method.
-        # TODO: This could be implemented more efficiently.
-        # It isn't too worrisome, because the heuristic handles most difficult
-        # cases.
+    """
+    Page, 251 section 7.3
+    """
+
+    C, K, trans_consts = base_Q(DE.D)
+    # trans_consts be t1, ..., tn
+    E_part = [frac_in(DE.D[i].quo(Poly(DE.T[i], DE.T[i])),
+        trans_consts) for i in DE.indices('exp')]
```

Comment:
An example would be
```python
In [4]: frac_in(x**2/(x + 1), [x, t, sqrt(2)])
Out[4]: (Poly(x**2, x, t, sqrt(2), domain='ZZ'), Poly(x + 1, x, t, sqrt(2), domain='ZZ'))
```
So it uses all the elements of list as generator for `num` and `den`.